docs/docs/reference/Stemming.md

```diff
@@ -6,14 +6,65 @@ description: >
     Stemming support
 ---
 
-# Stemming Support
+# Stemming and Multi-language Support
 
-RediSearch supports stemming - that is adding the base form of a word to the index. This allows the query for "going" to also return results for "go" and "gone", for example.
+RediSearch supports stemming - that is adding the base form of a word to the index. This allows the query for "`going`" to also return results for "`go`" and "`gone`", for example.
 
-The current stemming support is based on the Snowball stemmer library, which supports most European languages, as well as Arabic and other. We hope to include more languages soon (if you need a specific language support, please open an issue).
+The current stemming support is based on the Snowball stemmer library, which supports most European languages, as well as Arabic and other. See the "[Supported languages](#supported-languages)" section below. We hope to include more languages soon (if you need a specific language support, please open an issue).
 
 For further details see the [Snowball Stemmer website](http://snowballstem.org/).
 
+## How it works?
+
+Stemming maps different forms of the same word to a common root - "stem" - for example, the English stemmer maps *studied* ,*studies* and *study* to *stud* . So a searching for *studied* would also find documents which only have the other forms.
+
+In order to define which language the Stemmer should apply when building the index, you need to specify the `LANGUAGE` parameter for the entire index or for the specific field. For more details check the [FT.CREATE](/docs/commands/ft.create.md) syntax.
+
+**Create a index with language definition**
+
+Create a index for words in German "`wort:`" with a single `TEXT` field "`wort`"
+
+{{< highlight bash >}}
+redis> FT.CREATE idx:german ON HASH PREFIX 1 "wort:" LANGUAGE GERMAN SCHEMA wort TEXT
+{{< / highlight >}}
+
+**Adding words**
+
+Adding some words with same stem in German: `stück stücke stuck stucke` => `stuck`
```

Comment:
What if we add a note to clarify that ``Stücke`` is the plural form of ``Stück``?

Reply:
```suggestion
Adding some words with same stem in German, all variations of the word `stück` ( `piece` in english): `stück stücke stuck stucke` => `stuck`
```